crates/biome_css_parser/src/lexer/mod.rs

```diff
@@ -699,50 +672,96 @@ impl<'src> CssLexer<'src> {
         let mut idx = 0;
         let mut is_first = true;
         // Repeatedly consume the next input code point from the stream.
-        loop {
-            let Some(current) = self.current_byte() else {
+        while let Some(current) = self.current_byte() {
+            if let Some(part) = self.consume_ident_part(current, is_first) {
+                is_first = false;
+
+                if let Some(buf) = buf.get_mut(idx..idx + 4) {
```

Comment:
In this context, "+ 4" represents a safety measure for the buffer size.
It ensures that there are at least 4 slots available in the buffer.
This is necessary because the maximum UTF-8 sequence length is 4 bytes.